### PR TITLE
Speedup schema cache and contents __getattr__

### PR DESCRIPTION
**Pull request by jensens**
1. in schema cache, By first checking for string, the `providedBy` can be avoided. Since this function is called really often it makes a difference. According to my profiling its speeds up plone/dexterity/schema.py function decorator by factor ~1.54x.
2. in schema cache, the FTI itself can be cached on the request. Using the cache is twice as fast as the lookup itself.
3. in content `__getattr__` the BehaviorAssignable lookup is expensive and done very often. Do it once per type and save a lot of time. Speedup factor ~1.9x

If found this on my the hunt for the slowness of `unrestrictedTraverse`. A simple traversal to `/Plone/news/aggregator` repeating it a bunch of times was my test bed. 

TL;DR: 

profiling before:
```
10000 0.158 3.438 /home/jensens/.buildout/shared-eggs/Zope-4.1.1-py3.7.egg/OFS/Traversable.py:152(unrestrictedTraverse)
50002/30000 0.373 2.530 /home/jensens/workspace/cdev/plone5.2/src/plone.dexterity/plone/dexterity/content.py:333(__getattr__)
```
after:
```
10000 0.115 1.231 /home/jensens/.buildout/shared-eggs/Zope-4.1.1-py3.7.egg/OFS/Traversable.py:152(unrestrictedTraverse)
30003/30000 0.179 0.572 /home/jensens/workspace/cdev/plone5.2/src/plone.dexterity/plone/dexterity/content.py:336(__getattr__)
```

Overall performance gain in my profiling test: ~2.8x

### CODE REVIEW

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by gforcada** [comment]
If you want a speed up, avoid completely the `__getattr__` with @datakurre branch https://github.com/plone/plone.dexterity/tree/datakurre-no-getattr

**Comment by rafaelbco** [comment]
Just keep in mind that `__getattr__` is needed in order to have default values for fields.

**Comment by jensens** [comment]
will follow up with 2 new PRs: #114 and ...
